### PR TITLE
fix(terminal): prevent command concatenation in workspace panel display

### PR DESCRIPTION
**Pull request by bakedbean**
## Summary

Fixes a bug where running commands back-to-back in the terminal would display them concatenated in the workspace panel. For example, running `pnpm reset` followed by `pnpm dev` would show "pnpm resetpnpm dev" instead of just "pnpm dev".

## Root Cause

The OSC 133 shell integration parser wasn't clearing the command buffer or resetting tracking state when a new prompt appeared (OSC 133;A sequence). This allowed stale command text from interrupted or improperly terminated commands to persist and get concatenated with subsequent commands.

## Changes

- **OSC 133 parser (`src/osc133.rs`)**: Now clears command buffer and resets tracking flag when PromptStart (OSC 133;A) is received
- **PTY event handler (`src/pty.rs`)**: Now clears the stored command reference when a new prompt appears
- **Test coverage**: Added two test cases:
  - `test_consecutive_commands_not_concatenated`: Verifies consecutive commands don't concatenate
  - `test_command_buffer_cleared_on_prompt_start`: Verifies buffer clearing on prompt, even for interrupted commands

## Testing

- ✅ All 18 OSC 133 and PTY tests pass
- ✅ Clippy clean with no warnings
- ✅ Defensive fix handles edge cases where shells send OSC sequences in non-standard orders or commands are interrupted with Ctrl+C

## Risk Assessment

**Low risk** - The changes are defensive and additive:
- Only clears state when a new prompt appears (a natural reset point)
- Doesn't change existing logic for successful command completion
- Test coverage ensures no regressions in the OSC 133 parsing behavior

### CODE REVIEW

File: src-tauri/src/osc133.rs [hunk 1]
```diff
@@ -113,6 +113,9 @@ impl Osc133Parser {
         let s = String::from_utf8_lossy(&self.buffer);
 
         if s.starts_with("133;A") {
+            // Clear any stale command text when a new prompt appears
+            self.command_buffer.clear();
```
[the remark this answers is not in the document]
Comment (bakedbean) [reply src-tauri/src/osc133.rs]: Done — guarded the `command_buffer.clear()` on `PromptStart` behind a `tracking_command` check so that a `CommandExecuted` event earlier in the same `feed()` chunk retains its captured text until `extract_command()` is called. Added a regression test that feeds `133;C` + `133;A` in a single call to verify.

File: src-tauri/src/pty.rs [hunk 2]
```diff
@@ -160,11 +160,14 @@ pub async fn spawn_pty(
                                 );
                             }
                             Osc133Event::PromptStart => {
-                                // Prompt appeared - reset running state if still set
+                                // Prompt appeared - reset running state and clear stale command
                                 let mut running = running_clone.lock();
                                 if *running {
                                     *running = false;
                                 }
```
[the remark this answers is not in the document]
Comment (bakedbean) [reply src-tauri/src/pty.rs]: Done — each mutex is now locked and dropped independently (no nested guards), and the conditional was removed since assigning `false` is idempotent.
